src/Server.cpp

```diff
@@ -1015,6 +1010,8 @@ void handleCompleteRequestHeader(Server& server, int clientFd, Connection& conne
 	if (connection.m_request.httpStatus == StatusOK && connection.m_request.hasBody) {
 		connection.m_status = Connection::ReceiveBody;
 		connection.m_buffer.erase(0, connection.m_buffer.find("\r\n\r\n") + 4);
+		if (!connection.m_buffer.empty())
+			handleBody(server, clientFd, connection);
```

Comment:
The test ``POSTRequest`` from the ``HandleCompleteRequestHeaderTest`` test suite fails. 

This is the case because the test checks, if the ``m_connection.m_status`` equals ``Connection::ReceiveBody``, but the function ``handleBody`` sets it to ``Connection::BuildResponse``. 
Which is the result of the passed checks, that the body is complete and the request does not use CGI. 

Therefore this test should be adapted to the new implementation.

Reply:
Well recognised! The new procedure processes the body immediately without waiting to receive it, does so and then continues to build the response. Test adapted